packages/xo-server-load-balancer/src/index.js

```diff
@@ -94,63 +92,26 @@ export const configurationSchema = {
 
 // ===================================================================
 
-// Create a job not enabled by default.
-// A job is a cron task, a running and enabled state.
-const makeJob = (cronPattern, fn) => {
-  const job = {
-    running: false,
-    emitter: new EventEmitter(),
-  }
-
-  job.cron = new CronJob(cronPattern, async () => {
-    if (job.running) {
-      return
-    }
-
-    job.running = true
-
-    try {
-      await fn()
-    } catch (error) {
-      console.error(
-        '[WARN] scheduled function:',
-        (error && error.stack) || error
-      )
-    } finally {
-      job.running = false
-      job.emitter.emit('finish')
-    }
-  })
-
-  job.isEnabled = () => job.cron.running
-
-  return job
-}
-
-// ===================================================================
-// ===================================================================
-
 class LoadBalancerPlugin {
   constructor (xo) {
     this.xo = xo
-    this._job = makeJob(
-      `*/${EXECUTION_DELAY} * * * *`,
-      this._executePlans.bind(this)
+
+    this._job = createSchedule(`*/${EXECUTION_DELAY} * * * *`).createJob(
+      async () => {
+        try {
+          await this._executePlans()
+        } catch (error) {
+          console.error(
+            '[WARN] scheduled function:',
+            (error && error.stack) || error
+          )
+        }
+      }
     )
   }
 
-  async configure ({ plans }) {
-    const job = this._job
-    const enabled = job.isEnabled()
-
-    if (enabled) {
-      job.cron.stop()
-    }
-
-    // Wait until all old plans stopped running.
-    if (job.running) {
-      await eventToPromise(job.emitter, 'finish')
-    }
+  async configure ({ plans }, { loaded }) {
+    this._job.stop()
```

Comment:
Why are you stopping the job here?

Reply:
My bad we don't need to restart the job.